posts/2022/02/making-gpus-accessible-to-pydata-ecosystem-via-array-api.md

```diff
@@ -0,0 +1,481 @@
+<!--
+.. title: Making GPUs accessible to PyData Ecosystem via Array API
+.. slug: making-gpus-accessible-to-pydata-ecosystem-via-array-api
+.. date: 2022-02-01 21:07:02 UTC-06:00
+.. author: Amit Kumar
+.. tags: GPU, SciPy, NumPy, CuPy, scikit-learn, scikit-image, Array API
+.. category: 
+.. link: 
+.. description: 
+.. type: text
+.. previewimage: /images/2022/02/array_api_workflow.svg
+-->
+
+
+
+GPUs have become an essential part of the scientific computing stack and with
+the advancement in the technology around GPUs and the ease of accessing a GPU
+on cloud or on-prem, it is in the best interest of the PyData community to spend
+time and effort to make GPUs accessible for the users of PyData libraries. A
+typical user in the PyData ecosystem is quite familiar with the APIs of libraries
+like scipy, scikit-learn, scikit-image and at the moment all these
+libraries only support single core operations on CPU. In this blog post will
+talk about how we can use Array API with the fundamental libraries in the PyData
+ecosystem along with CuPy for making GPUs accessible to the users of these libraries.
+With the introduction of the [Python Array API Standard](https://data-apis.org/array-api/latest/)
+by the [Consortium for Python Data API Standards](https://data-apis.org/) and it’s
+adoption mechanism in the [NEP 47](https://numpy.org/neps/nep-0047-array-api-standard.html)
+it is now possible to write code that is portable between NumPy and other array/tensor
+libraries that adopt the Array API Standard. We will also discuss the workflow and
+challenges for achieving the same.
+
+<!-- TEASER_END -->
+
+## Motivation and Goal
+
+The motivation for this exercise comes from the fact that having GPU-specific code
+in SciPy and scikits will be too-specialized and very hard to maintain. The ultimate
+goal is to have GPU support in SciPy, scikit-learn and scikit-image using CuPy. Since
```

Comment:
Note that using CuPy is not quite the ultimate goal. Rather, we want to enable these libraries to accept any type of array, so they can work with CuPy, PyTorch, JAX, Dask and other libraries. Then hardware vendors like AMD can focus on support for their own type of hardware.

More importantly, there's a mix here between the goal of the larger project and the goal of this demo. Describing both would be good.